java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -68,6 +72,17 @@ public RangeEqualsVisitor(ValueVector left, ValueVector right, boolean isTypeChe
     }
   }
 
+  /**
+   * Validate the passed left vector, if it is changed, reset and check type.
+   */
+  protected boolean validate(ValueVector left) {
+    if (left != this.left) {
+      this.left = left;
+      checkType();
+    }
+    return typeCompareResult;
```

Comment:
I think this way (a member boolean typeCompareResult) makes the code less readable. I am wondering why not make checkType method look like this:

boolean void checkType()

Reply:
agree with @liyafan82

Reply:
@liyafan82  @pravindra Hmm, this member was just introduced by your previous PR. I think the reason is that cache this result in constructor and reused in rangeEqualsAPI each time.
See if we have good solution.

Reply:
@tianchen92 I see. Thanks for the clarification.

Reply:
thanks for the clarification.